src/checkers/inference/DefaultSlotManager.java

```diff
@@ -359,6 +370,23 @@ public ExistentialVariableSlot createExistentialVariableSlot(VariableSlot potent
         return existentialVariableSlot;
     }
 
+    @Override
+    public ArithmeticVariableSlot createArithmeticVariableSlot(AnnotationLocation location) {
+        if (location == null || location == AnnotationLocation.MISSING_LOCATION) {
+            ErrorReporter
+                    .errorAbort("Cannot create an ArithmeticVariableSlot with a missing location");
+        }
+
+        // create the arithmetic var slot if it doesn't exist for the given location
+        if (!arithmeticSlotCache.containsKey(location)) {
+            ArithmeticVariableSlot slot = new ArithmeticVariableSlot(location, nextId());
```

Comment:
Why not declared `slot` outside this if block, so that if a new `ArithmeticVariableSlot` is created, we directly return its reference without looking up the storage again?

Reply:
The other possible problem is similar to `VariableSlot`: two `MISSING_LOCATION`s might be different. But in your implementation, they are treated the same, and it always returns a unique `ArithmeticVariableSlot` for that. If `getCurrentLocation()` returns `MISSING_LOCATION` for some reason for two different trees, it won't make sense any more to use only one slot.

Reply:
> Why not declare slot outside this if block, so that if a new ArithmeticVariableSlot is created, we directly return its reference without looking up the storage again? Or we could directly return at the end of this if block.

Returning directly.

> The other possible problem is similar to VariableSlot: two MISSING_LOCATIONs might be different. But in your implementation, they are treated the same, and it always returns a unique ArithmeticVariableSlot for that. 

See Line 375 above.